client/merchant-feedback-prompt/hooks.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * External dependencies
+ */
+import { useUserPreferences } from '@woocommerce/data';
+
+/**
+ * Extend the user preferences returned from useUserPreferences to include the WooPayments merchant feedback prompt dismissed state.
+ * See WC_Payments::add_user_data_fields() in includes/class-wc-payments.php for the PHP implementation.
+ */
+interface UserPreferences extends ReturnType< typeof useUserPreferences > {
+	/** The unix timestamp of the dismissal of the merchant feedback prompt. */
+	wc_payments_wporg_review_2025_prompt_dismissed?: number;
+}
+
+/**
+ * A hook for managing the merchant feedback prompt visibility state.
+ * It returns the current visibility state and a function to update the state.
+ */
+export const useMerchantFeedbackPromptState = () => {
+	const {
+		updateUserPreferences,
+		...userPrefs
+	} = useUserPreferences() as UserPreferences;
+
+	const isAccountEligible =
+		wcpaySettings?.featureFlags?.isMerchantFeedbackPromptDevFlagEnabled &&
+		wcpaySettings?.accountStatus?.campaigns?.wporgReview2025;
```

Comment:
As mentioned in https://github.com/Automattic/woocommerce-payments/pull/10370/files#r1976770261, consider using early return logic to make this easier to read.